File: Server.Core/Server.Core.Lists/Models/ListItemModel.cs
using System;

namespace Server.Core.Lists.Models
{
    /// <summary>
    /// Модель пункта списка.
    /// </summary>
    public class ListItemModel
    {
        /// <summary>
        /// Код пункта списка.
        /// </summary>
        public string Id { get; set; }

        /// <summary>
        /// Содержание списка.
        /// </summary>
        public string Content { get; set; }

        /// <summary>
        /// Позиция в пункте меню.
        /// </summary>
        public int OrderPosition { get; set; }

        /// <summary>
        /// Дата создания UTC.
        /// </summary>
        public DateTime CreateEventTime { get; set; }

        /// <summary>
        /// Дата редактирования UTC.
        /// </summary>
        public DateTime EditEventTime { get; set; }

        /// <summary>
        /// Количество лайков.
        /// </summary>
        public int LikeCount { get; set; }

        /// <summary>
        /// Код владельца.
        /// </summary>
        public string OwnerId { get; set; }

        /// <summary>
        /// Признак выбранного пункта.
        /// </summary>
        public bool IsChecked { get; set; }
    }
}

File: Client/SpiskerApp/ClientApp/app/Lists/Models/ListItemModel.ts
export class ListItemModel {

    id: string;

    content: string;

    orderPosition: number;

    createEventTime: Date;

    editEventTime: Date;

    likeCount: number;

    ownerId: string;

    isInserted: boolean;

    isChecked: boolean;

    data: any;

    static compareByPosition(a: ListItemModel, b: ListItemModel):number {
        return a.orderPosition - b.orderPosition;
    }
}